### ISSUE TITLE
`CloneIn` set `SymbolId`, `ScopeId`, `ReferenceId` to `None`

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#4809]
As discussed in #4804, we feel that it's dangerous for `CloneIn` to clone `scope_id`, `symbol_id` and `reference_id` fields, as they're going to be incorrect. It should set them to `None` instead.

**Comment by rzvxa** [comment]
Shall I submit a PR for it?

**Comment by overlookmotel** [comment]
If you have time, that'd be great!

**Comment by rzvxa** [comment]
Yeah, I'd like to take this one. I'm working on codegen markers right now and we can express this cleanly through a marker.

I have something like this in mind:

```rust
#[ast]
#[generate_derive(CloneIn)]
pub struct Struct<'a> {
    pub a: Span,
    pub b: Vec<'a, Statement<'a>>,
    // instead of deriving a `CloneIn::clone_in` for this field we use `Default::default`
    #[clone_in(never)] // or #[clone_in(skip)]?
    pub c: Cell<Option<ScopeId>>,
}
```

I find this much better than having edge cases for these fields, It lets the users know clearly that these fields aren't cloned with `clone_in`.

**Comment by overlookmotel** [comment]
Nice! How about `#[clone_in(default)]` since the field will still exist in the clone, but will be set to default value (`None`)?

**Comment by rzvxa** [comment]
Honestly, I can't remember why I thought default wouldn't be a good choice here. It had something to do with how we might want to expose an unsafe method to skip these checks.
I'll go with `clone_in(default)` if I can remember the issue I had with it I'll come back to you for comments, Otherwise, it means I was just imagining stuff😅.

**Comment by overlookmotel** [comment]
Implemented in #4819.
